### PR TITLE
Get blueprint namespace directly from blueprintjs

### LINKED ISSUE #181: React mosaic 5.0 with Blueprintjs 4.0 has window buttons that are too big
### Bug report

- __Package version(s)__: 5.0.0
- __Browser and OS versions__: Chrome 99, Windows 11

#### Steps to reproduce

When using react-mosaic version 5.0.0 with @blueprintjs/core 4.0.0, the buttons seem to be too big for the toolbarL

image

#### Actual behavior

You can see it reproduced in the codesandbox here:

https://codesandbox.io/s/react-mosaic-blueprint-4-cxfdvg

#### Notes

The behavior only exhibits itself if blueprintjs *and* react-mosaic are at the versions above; if blueprint js is at version 3.54.0 or react-mosaic is at version 4.1.1 the height of the buttons fits the toolbar.

Looking at the rendered output, it seems like the buttons are using the `bp3` prefix:

image

Even though `bp4` is the recommendation for blueprint version 4:

https://blueprintjs.com/docs/#core/classes.namespacing

I'm not sure if there's a way to do this already in the project, but it seems that it needs to be "aware" of blueprint 4.

When changing the prefixes (through the browser) to bp4, the buttons seem to be properly proportioned.

### PR DESCRIPTION
**Pull request by casperOne**
#### Changes proposed in this pull request:

Update `OptionalBlueprint` to set the blueprint namespace constant using a from the `Classes` type exported from blueprint.

Confirmed that it is available in both 3.0 and 4.0:

https://github.com/palantir/blueprint/blob/develop/packages/core/src/common/classes.ts#L316
https://github.com/palantir/blueprint/blob/%40blueprintjs/core%403.54.0/packages/core/src/common/classes.ts#L322

#### Reviewers should focus on:

The buttons being sized correctly in react mosaic when using blueprintjs 4.0 and react mosaic 5.0

### CODE REVIEW

File: src/util/OptionalBlueprint.tsx [hunk 1]
```diff
@@ -1,36 +1,11 @@
-import type { Classes } from '@blueprintjs/core';
+import { Classes } from '@blueprintjs/core';
 import type { IconNames } from '@blueprintjs/icons';
 import classNames from 'classnames';
 import _ from 'lodash';
 import * as React from 'react';
 
-// Performs a top level import of blueprint and gets the classes.
-async function getBluePrintClasses() {
```
Comment (nomcopter) [inline src/util/OptionalBlueprint.tsx]: Keeping it optional would be nice - why did you end up removing this? Lack of support in build tools?

Comment (casperOne) [reply src/util/OptionalBlueprint.tsx]: Top level async statements are not supported when the target is `es2017`; the library would have to be bumped up to `es2020` in order to take advantage of that.

Without top level awaits (and support for `import` to be called as a function), this approach wouldn't work.

If you're willing to have the module/lib updated to `es2020`, I'd be happy to make that change and include this back into the PR.

**Comment by casperOne** [comment]
I'm going to pull this back and create a clean PR which addresses the bp3 issue.

If you'd like a PR around loading this dynamically, I'll do that, separately, but it will require a lot of upgrading to the build tooling, which obfuscates the issue.

**Comment by nomcopter** [comment]
I appreciate all the action here - I'm working on adding `blueprintNamespace` as a prop to `Mosaic` which should make this possible while still being able to target Blueprint v4

**Comment by nomcopter** [comment]
Should have a release today

**Comment by casperOne** [comment]
@nomcopter appreciated, thanks!

TBH, I like this solution, allowing us to define it in code, but I wasn't sure if that was in the spirit that you intend for the library

**Comment by nomcopter** [comment]
Yeah it would be nice to do it automatically with proper optional dependencies or by just requiring blueprint but build tooling support for the former is obnoxious and restrictive downstream and the latter complicates things for those that don't want it. Appreciate the thorough issues and PR!

**Pull request closed without merging**